packages/dds/tree/src/util/typeUtils.ts

```diff
@@ -157,15 +157,33 @@ export type _RecursiveTrick = never;
  * Alternative to the built in Record type which does not permit unexpected members,
  * and is readonly.
  *
+ * @remarks
+ * This does not work correctly when `K` is more specific than `string` or `symbol`.
+ * For example `{a: 5}` is not assignable to `RestrictiveReadonlyRecord<"a",: number>`
+ *
  * @privateRemarks
  * `number` is not allowed as a key here since doing so causes the compiler to reject recursive schema.
  * The cause for this is unclear, but empirically it was the case when this comment was written.
+ *
+ * @deprecated Use a more robust / specific type instead which actually works.
```

Comment:
Nit: what about something like
```suggestion
 * @deprecated Use a more robust / specific type instead. This type never worked as intended.
```